main/server/src/mill/main/server/Server.scala

```diff
@@ -48,7 +46,8 @@ abstract class Server[T](
 
         while (
           running && {
-            val serverSocket = bindSocket()
+            val serverSocket = new java.net.ServerSocket(0)
```

Comment:
This binds to 0.0.0.0 for me, so it is listening to the world, including external network. At least on my machine.

So, better IMO:
```
        var serverSocket = new java.net.ServerSocket();
        serverSocket.bind(new java.net.InetSocketAddress("127.0.0.1",0));
```

I suggesting  the IP 127.0.0.1 over 'localhost', because localhost sometimes resolves to IP6 vs IP4, creating odd miss-matches.